### PR TITLE
Fix internal `default_hash_types` lint to use resolved path

### PR DESCRIPTION
**Pull request by camsteffen**
I run into false positives now and then (mostly in Clippy) when I want to name some util after HashMap.

### CODE REVIEW

File: src/tools/clippy/clippy_lints/src/implicit_hasher.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// todo remove after fix is synced
 #![allow(rustc::default_hash_types)]
```
Comment (flip1995) [inline src/tools/clippy/clippy_lints/src/implicit_hasher.rs:2]: Not 100% sure, but removing this during sync might fail the rust-lang/rust CI afterwards. Shouldn't this be changed to `cfg_attr(bootstrap, ...)`?

**Comment by camsteffen** [comment]
Reverted removal of allow in rustc_macros

**Pull request merged**